rfcs/text/0016_buildkite.md

```diff
@@ -0,0 +1,907 @@
+      "allowed_list": ["renovate[bot]"],
+      "set_commit_status": true,
+      "commit_status_context": "kibana-buildkite",
+      "trigger_comment_regex": "^(?:(?:buildkite\\W+)?(?:build|test)\\W+(?:this|it))|^retest$"
+    }
+  ]
+}
+```
+
+Github Webhooks must also be configured to send events to the deployed bot.
+
+#### Build / Deploy
+
+Currently, the bot is built and deployed using [Google Cloud Build](https://cloud.google.com/build). It is deployed to and hosted on [Google Cloud Run](https://cloud.google.com/run). It uses [Google Secret Manager](https://cloud.google.com/secret-manager) for storing/retrieving tokens for accessing GitHub and Buildkite.
+
+TODO link to cloud build yaml once available
+
+### Infrastructure
+
+We will need to maintain our infrastructure related to Buildkite, primarily ephemeral agents. To start, it will mean supporting infrastructure in GCP, but could later mean AWS as well.
+
+- Separate GCP project for CI resources
+- Hosting for bots/services we maintain, such as the Agent Manager (GKE Auto-Pilot) and GitHub PR bot (Cloud Run)
+- Google Storage Buckets for CI artifacts
+- Networking (security, we may also need Cloud NAT)
+- IAM and Security
+- Agent images
+
+We are already using Terraform to manage most resources related to Buildkite, and will continue to do so.
+
+### Monitoring / Alerting
+
+We will need to set up and maintain monitoring and alerting for our GCP infrastructure, as well as Buildkite metrics.
+
+Some examples:
+
+GCP
+
+- Number of instances by type
+- Age of instances
+- Resource Quotas
+
+Buildkite
+
+- Agent queues
+- Job wait times
+- Build status
+
+### Agent Image management
+
+We will need to maintain images used to create GCP instances for our Buildkite agents. These images would need to be built on a regular basis (daily, or possibly more often).
+
+We could likely maintain a single linux-based image to cover all of our current CI needs. However, in the future, if we need to maintain many images across different operating systems and architectures, this is likely to become the most complex part of the CI system that we would need to maintain. Every operating system and architecture we need to support adds another group of required images, with unique dependencies and configuration automation.
+
+Another thing to note: Just because we need to run something on a specific OS or architecture, it doesn't necessarily mean we need to maintain an agent image for it. For example, we might use something like Vagrant to create a separate VM, using the default, cloud-provided images, that we run something on (e.g. for testing system packages), rather than running it on the same machine as the agent. In this case, we would potentially only be managing a small number of images, or even a single image.
+
+For our testing, we have a single GCP image, built using Packer, with the Buildkite agent installed and all of our dependencies.
+
+Summary of Responsibilities
```

Comment:
Ideally, we would also want to test against this new image similar to what we do with ES Snapshots. But this is not something we do today.